torch/optim/adagrad.py

```diff
@@ -28,9 +29,11 @@ def __init__(self, params, lr=1e-2, lr_decay=0, weight_decay=0, initial_accumula
             raise ValueError("Invalid weight_decay value: {}".format(weight_decay))
         if not 0.0 <= initial_accumulator_value:
             raise ValueError("Invalid initial_accumulator_value value: {}".format(initial_accumulator_value))
+        if not 0.0 <= eps:
```

Comment:
You also need to implement `__setstate__` to add this attribute, because it will be missing in existing model pickles.

Reply:
Shall this time be okay?